hooks/pre-push.d/eof-newline-test

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+top=`git rev-parse --show-toplevel`
+status=0
+for file in $(find . -path ./target -prune -o -type f -name "*.rs" -o -name "*.toml"); do
+    if [ -f "$file" ] && [ ! -z $(tail -c 1 "$file") ]; then
```

Comment:
```suggestion
    if [ ! -z $(tail -c 1 "$file") ]; then
```

Also, this tests for any whitespace, not specifically newline.

Reply:
Right. Fixed in latest push.